webapp/app/[locale]/tunnel/_hooks/useDeposit.ts

```diff
@@ -1,303 +1,221 @@
-    enabled: !depositingNative && canDeposit,
-    extendedApproval: extendedErc20Approval,
-    fromToken,
-    onApprovalSuccess,
-    onSuccess: onUserAcceptingDeposit,
-    toToken,
-  })
-
-  const {
-    data: depositReceipt,
-    error: depositReceiptError,
-    queryKey: depositQueryKey,
-  } = useWaitForTransactionReceipt({
-    hash: depositingNative ? depositNativeTokenTxHash : depositErc20TokenTxHash,
-  })
-
-  useReloadBalances({
-    fromToken,
-    status: depositReceipt?.status,
-  })
-
-  const clearDepositNativeState = useCallback(
-    function () {
-      // clear the deposit operation hash
-      resetDepositNativeToken()
-      // clear transaction receipt state
-      queryClient.removeQueries({ queryKey: depositQueryKey })
-    },
-    [depositQueryKey, queryClient, resetDepositNativeToken],
+  const { track } = useUmami()
+  const updateNativeBalanceAfterFees = useUpdateNativeBalanceAfterReceipt(
+    fromToken.chainId,
   )
+  const { l1WalletClient } = useL1WalletClient(fromToken.chainId)
 
-  const clearDepositTokenState = useCallback(
-    function () {
-      // clear the approval operation hash, if any
-      resetApproval?.()
-      // clear the deposit operation hash
-      resetDepositToken()
-      // clear approval receipt state
-      queryClient.removeQueries({ queryKey: approvalQueryKey })
-      // clear transaction receipt state
-      queryClient.removeQueries({ queryKey: depositQueryKey })
-    },
-    [
-      approvalQueryKey,
-      depositQueryKey,
-      queryClient,
-      resetApproval,
-      resetDepositToken,
-    ],
-  )
+  const depositingNative = isNativeAddress(fromToken.address)
 
-  useEffect(
-    function updateDepositStatusAfterFailure() {
-      if (!depositReceiptError) {
-        return
-      }
-      const deposit = deposits.find(
-        d =>
-          d.transactionHash === currentTxHash &&
-          d.status === EvmDepositStatus.DEPOSIT_TX_PENDING,
-      )
-      if (!deposit) {
-        return
-      }
-      clearDepositNativeState()
-      clearDepositTokenState()
+  const { allowanceQueryKey } = useNeedsApproval({
+    address: fromToken.address,
+    amount,
+    spender: l1StandardBridgeAddress,
+  })
 
-      updateDeposit(deposit, {
-        status: EvmDepositStatus.DEPOSIT_TX_FAILED,
+  return useMutation({
+    mutationFn() {
+      track?.('evm - dep started', { chain: networkType })
+
+      const { promise, emitter } = depositingNative
+        ? depositEth({
+            account: address,
+            amount,
+            // here both chains are always EVM
+            l1Chain: findChainById(fromToken.chainId) as Chain,
+            l1PublicClient: getEvmL1PublicClient(fromToken.chainId),
+            l1WalletClient,
+            l2Chain: findChainById(toToken.chainId) as Chain,
+          })
+        : depositErc20({
+            account: address,
+            amount,
+            approvalAmount:
+              !depositingNative && extendedErc20Approval
```

Comment:
`depositingNative` is always false here.

Reply:
Indeed, removed [0b23631](https://github.com/hemilabs/ui-monorepo/pull/1125/commits/0b236310ab704fd5877937ec4c6164ba85633df7)